include/libopencm3/stm32/g0/rcc.h

```diff
@@ -0,0 +1,948 @@
+#define RCC_CFGR_MCOPRE_DIV2	    1
+#define RCC_CFGR_MCOPRE_DIV4	    2
+#define RCC_CFGR_MCOPRE_DIV8	    3
+#define RCC_CFGR_MCOPRE_DIV16	    4
+#define RCC_CFGR_MCOPRE_DIV32	    5
+#define RCC_CFGR_MCOPRE_DIV64	    6
+#define RCC_CFGR_MCOPRE_DIV128	    7
+/*@}*/
+
+#define RCC_CFGR_MCO_SHIFT			24
+#define RCC_CFGR_MCO_MASK			0xf
+
+/** @defgroup rcc_cfgr_mcosel MCO Sel
+ * @ingroup STM32G0xx_rcc_defines
+ * @brief Microcontroler clock output selector
+@sa rcc_cfgr_mcosel
+@{*/
+#define RCC_CFGR_MCO_NOCLK			0x0
+#define RCC_CFGR_MCO_SYSCLK			0x1
+#define RCC_CFGR_MCO_HSI16			0x3
+#define RCC_CFGR_MCO_HSE			0x4
+#define RCC_CFGR_MCO_PLLRCLK		0x5
+#define RCC_CFGR_MCO_LSI			0x6
+#define RCC_CFGR_MCO_LSE			0x7
+/*@}*/
+
+#define RCC_CFGR_PPRE_MASK		0x7
+#define RCC_CFGR_PPRE_SHIFT		12
+
+/** @defgroup rcc_cfgr_ppre PPRE
+ * @ingroup STM32G0xx_rcc_defines
+ * @brief APB Prescaler
+@sa rcc_cfgr_ppre
+@{*/
+#define RCC_CFGR_PPRE_NODIV		0x0
+#define RCC_CFGR_PPRE_DIV2		0x4
+#define RCC_CFGR_PPRE_DIV4		0x5
+#define RCC_CFGR_PPRE_DIV8		0x6
+#define RCC_CFGR_PPRE_DIV16		0x7
+/*@}*/
+
+#define RCC_CFGR_HPRE_MASK		0xf
+#define RCC_CFGR_HPRE_SHIFT		8
+/** @defgroup rcc_cfgr_hpre HPRE
+ * @ingroup STM32G0xx_rcc_defines
+ * @brief APB Prescaler
+@sa rcc_cfgr_hpre
+@{*/
+#define RCC_CFGR_HPRE_NODIV		0x0
+#define RCC_CFGR_HPRE_DIV2		0x8
+#define RCC_CFGR_HPRE_DIV4		0x9
+#define RCC_CFGR_HPRE_DIV8		0xa
+#define RCC_CFGR_HPRE_DIV16		0xb
+#define RCC_CFGR_HPRE_DIV64		0xc
+#define RCC_CFGR_HPRE_DIV128	0xd
+#define RCC_CFGR_HPRE_DIV256	0xe
+#define RCC_CFGR_HPRE_DIV512	0xf
+/*@}*/
+
+#define RCC_CFGR_SWS_MASK			0x3
+#define RCC_CFGR_SWS_SHIFT			3
+/** @defgroup rcc_cfgr_sws SWS
+ * @ingroup STM32G0xx_rcc_defines
+ * @brief System clock switch status
+@sa rcc_cfgr_sws
+@{*/
+#define RCC_CFGR_SWS_HSISYS			0x0
+#define RCC_CFGR_SWS_HSE			0x1
+#define RCC_CFGR_SWS_PLLRCLK		0x2
+#define RCC_CFGR_SWS_LSI			0x3
+#define RCC_CFGR_SWS_LSE			0x4
+/*@}*/
+
+#define RCC_CFGR_SW_MASK			0x3
+#define RCC_CFGR_SW_SHIFT			0
+/** @defgroup rcc_cfgr_sws SW
+ * @ingroup STM32G0xx_rcc_defines
+ * @brief System clock switch
+@sa rcc_cfgr_sw
+@{*/
+#define RCC_CFGR_SW_HSISYS			0x0
+#define RCC_CFGR_SW_HSE				0x1
+#define RCC_CFGR_SW_PLLRCLK			0x2
+#define RCC_CFGR_SW_LSI				0x3
+#define RCC_CFGR_SW_LSE				0x4
+/*@}*/
+
+/* --- RCC_PLLCFGR - PLL Configuration Register */
+
+#define RCC_PLLCFGR_PLLR_SHIFT		29
+#define RCC_PLLCFGR_PLLR_MASK		0x7
+/** @defgroup rcc_pllcfgr_pllr PLLR
+ * @ingroup STM32G0xx_rcc_defines
+ * @brief VCO Division factor R for PLLRCLK clock output [2..8]. Frequency must not exceed 64mhz in voltage range 1, or 16mhz in voltage range 2.
+@sa rcc_pllcfgr_pllr
+@{*/
+#define RCC_PLLCFGR_PLLR_DIV(x)		((x)-1)
+#define RCC_PLLCFGR_PLLR_DIV2		1
+#define RCC_PLLCFGR_PLLR_DIV3		2
+#define RCC_PLLCFGR_PLLR_DIV4		3
+#define RCC_PLLCFGR_PLLR_DIV5		4
+#define RCC_PLLCFGR_PLLR_DIV6		5
+#define RCC_PLLCFGR_PLLR_DIV7		6
+#define RCC_PLLCFGR_PLLR_DIV8		7
```

Comment:
do we really need both versions?  there's no backwards compat here.

Reply:
Indeed. I prefer the macro but current code base has both. I'll remove them.